File: ASP.NET Web Forms Application Using Entity Framework Database First - V2/[VB]-ASP.NET Web Forms Application Using Entity Framework Database First - V2/VB/ContosoUniversity/DAL/Department.vb
Imports System.ComponentModel
Imports System.ComponentModel.DataAnnotations

<MetadataType(GetType(DepartmentMetaData))>
Partial Public Class Department
End Class

Public Class DepartmentMetaData
    <DataType(DataType.Currency), Range(0, 1000000, ErrorMessage:="Budget must be less than $1,000,000.00")>
    Public Property Budget() As Decimal

    <DisplayFormat(DataFormatString:="{0:d}", ApplyFormatInEditMode:=True)>
    Public Property StartDate() As Date

End Class

File: ASP.NET Web Deployment using Visual Studio/[C#]-ASP.NET Web Deployment using Visual Studio/C#/ContosoUniversity.DAL/Department.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ContosoUniversity.DAL
{
    public class Department
    {
        public Department()
        {
            Courses = new List<Course>();
        }

        public int DepartmentID { get; set; }

        [Required(ErrorMessage = "Department name is required.")]
        [MaxLength(50)]
        public string Name { get; set; }

        [DisplayFormat(DataFormatString = "{0:c}")]
        [Required(ErrorMessage = "Budget is required.")]
        [Column(TypeName = "money")]
        public decimal? Budget { get; set; }

        [DisplayFormat(DataFormatString = "{0:d}", ApplyFormatInEditMode = true)]
        [Required(ErrorMessage = "Start date is required.")]
        public DateTime StartDate { get; set; }

        [Display(Name = "Administrator")]
        public int? PersonID { get; set; }
        
        public virtual Instructor Administrator { get; set; }
        public virtual ICollection<Course> Courses { get; set; }
    }
}